src/userprog/process.c

```diff
@@ -28,92 +31,274 @@ static bool load (const char *cmdline, void (**eip) (void), void **esp);
 tid_t
 process_execute (const char *file_name) 
 {
-  char *fn_copy;
+  char *fn_copy=NULL, *argvs, *program=NULL;
+  struct tcb *tcb=NULL;
   tid_t tid;
 
-  /* Make a copy of FILE_NAME.
-     Otherwise there's a race between the caller and load(). */
+  //lock_acquire(&thread_current()->child_lock);
   fn_copy = palloc_get_page (0);
-  if (fn_copy == NULL)
-    return TID_ERROR;
-  strlcpy (fn_copy, file_name, PGSIZE);
+  program = palloc_get_page (0);
+  tcb = palloc_get_page(0);
+  //fn_copy = calloc(1,sizeof*fn_copy);
+  //program = calloc(1,sizeof*program);
+  //tcb = calloc(1, sizeof*tcb);
+
+  if(tcb == NULL || fn_copy == NULL || program == NULL)
+    PANIC("NULL NULL NULL!!");
 
-  /* Create a new thread to execute FILE_NAME. */
-  tid = thread_create (file_name, PRI_DEFAULT, start_process, fn_copy);
-  if (tid == TID_ERROR)
-    palloc_free_page (fn_copy); 
+  strlcpy (fn_copy, file_name, PGSIZE);
+  strlcpy (program, file_name, PGSIZE); 
+  program = strtok_r(program, " ",&argvs);
```

Comment:
Copy our file_name to access the tokenize it into program. program is a list now